marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs

```diff
@@ -568,12 +568,19 @@ class FlowUtils {
     return headers;
   }
 
-  mergeHeaders(headers, docHeaders, outputFormat) {
+  /*
+  This function will return the header obtained by merging "headersFromStepOptions" into "docHeaders",
+  where the former will overwrite the latter where there are duplicate keys.
+   */
+  mergeHeaders(docHeaders, headersFromStepOptions, outputFormat) {
     if (outputFormat === this.consts.XML) {
-      headers = this.cleanData(Sequence.from([
-        docHeaders,
-        this.jsonToXml(headers)
-      ]), "headers", outputFormat);
+      if(docHeaders instanceof Element || docHeaders instanceof Sequence){
+        headersFromStepOptions = this.mergeXmlHeaders(Sequence.from([docHeaders, this.jsonToXml(headersFromStepOptions)]));
+      }
+      else{
+        headersFromStepOptions = this.mergeXmlHeaders(Sequence.from([this.jsonToXml(docHeaders), this.jsonToXml(headersFromStepOptions)]));
```

Comment:
@ryanjdew ,
Should we call this.cleanData(this.mergeXmlHeaders(Sequence.from([this.jsonToXml(docHeaders), this.jsonToXml(headersFromStepOptions)]))) here or is this sufficient ?